### PR TITLE
SNOW-1437846: Enforce to_timestamp_ntz in resample

### PR DESCRIPTION
**Pull request by sfc-gh-lmukhopadhyay**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1437846

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Enforces all datetimes to be converted to TIMESTAMP_NTZ as timestamps with timezones are not supported in Snowpark pandas resample

### CODE REVIEW

**Review by sfc-gh-nkrishna — APPROVED** [review]

File: tests/integ/modin/resample/test_resample.py [hunk 1]
```diff
@@ -65,6 +65,26 @@ def test_resample_date_before_snowflake_alignment_date():
     )
 
 
+@sql_count_checker(query_count=2, join_count=1)
+def test_resample_date_general_ntz():
+    date_data = native_pd.to_datetime(
+        [
+            "2018-12-01",
```
Comment (sfc-gh-azhan) [inline tests/integ/modin/resample/test_resample.py:72]: What happened if the index to be resampled contains tz info?
cc @sfc-gh-nkrishna

File: tests/integ/modin/resample/test_resample.py [hunk 2]
```diff
@@ -65,6 +65,26 @@ def test_resample_date_before_snowflake_alignment_date():
     )
 
 
+@sql_count_checker(query_count=2, join_count=1)
+def test_resample_date_general_ntz():
```
Comment (sfc-gh-nkrishna) [inline tests/integ/modin/resample/test_resample.py:69]: @sfc-gh-lmukhopadhyay We already have this test `test_resample_date_wraparound_snowflake_alignment_date` so I think it's okay not to have this

File: tests/integ/modin/resample/test_resample.py (hunk 1 above)
Comment (sfc-gh-lmukhopadhyay) [reply tests/integ/modin/resample/test_resample.py:72]: Testcase 
```date_data = native_pd.to_datetime(
        [
            '2014-08-01 09:00:00+02:00',
            '2014-08-01 10:00:00+02:00',
            '2014-08-01 11:00:00+02:00'
        ]
    )
    eval_snowpark_pandas_result(
        *create_test_dfs({"A": np.random.randn(3)}, index=date_data),
        lambda df: df.resample("2D").min(),
        check_freq=False,
    )
```
is returning  `TypeError: Cannot subtract tz-naive and tz-aware datetime-like objects.` We may need better error messaging but we do not support resample with timezone right?

Comment (sfc-gh-nkrishna) [reply tests/integ/modin/resample/test_resample.py:72]: Resample currently does not work with tz info, we get `TypeError: Cannot subtract tz-naive and tz-aware datetime-like objects.`

Comment (sfc-gh-nkrishna) [reply tests/integ/modin/resample/test_resample.py:72]: @sfc-gh-lmukhopadhyay Can we add the test case you described above to `test_resample_negative` as part of this PR?

File: tests/integ/modin/resample/test_resample.py (hunk 2 above)
Comment (sfc-gh-lmukhopadhyay) [reply tests/integ/modin/resample/test_resample.py:69]: That's true I'll remove this

**Review by sfc-gh-azhan — APPROVED** [review]

**Review by sfc-gh-nkrishna — APPROVED** [review]

**Pull request merged**